crates/ruff_linter/src/rules/pylint/rules/magic_value_comparison.rs

```diff
@@ -57,39 +57,39 @@ impl Violation for MagicValueComparison {
     }
 }
 
-/// If an [`Expr`] is a literal (or unary operation on a literal), return the literal [`Expr`].
-fn as_literal(expr: &Expr) -> Option<&Expr> {
+/// If an [`Expr`] is a literal (or unary operation on a literal), return the [`LiteralExpressionRef`].
+fn as_literal(expr: &Expr) -> Option<LiteralExpressionRef<'_>> {
```

Comment:
nit: make this a method on `LiteralExpressionRef`?

Reply:
This method seems to be specific to the rule implementation (e.g. it unwraps unary expressions)

Reply:
Yes, it's specific to this rule. It basically unwraps a `Expr` into `LiteralExpressionRef` but includes a unary operation on a literal as well.